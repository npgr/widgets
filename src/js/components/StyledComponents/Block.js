import styled from "styled-components";
import PropTypes from "prop-types";

const Block = styled.div`
    ${({ align }) => align && `text-align: ${align};`}
    ${({ bgColor }) => bgColor && `background-color: ${bgColor};`}
    ${({ flex }) => flex && "display: flex;"}
    ${({ height }) => height && `height: ${height};`}
    ${({ pright }) => pright && `padding-right: ${pright};`}
    ${({ ptop }) => ptop && `padding-top: ${ptop};`}
    ${({ spaceBetween }) => spaceBetween && "justify-content: space-between;"}
    ${({ width }) => width && `width: ${width};`}
    padding: ${({ padding }) => padding || "2vh 2vw"};
`;

Block.propTypes = {
  align: PropTypes.string,
  bgColor: PropTypes.string,
  flex: PropTypes.bool,
  height: PropTypes.string,
  padding: PropTypes.string,
  pright: PropTypes.string,
  ptop: PropTypes.string,
  spaceBetween: PropTypes.bool,
  width: PropTypes.string
};

export default Block;
