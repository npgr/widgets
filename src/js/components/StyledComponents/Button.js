import styled from 'styled-components'
import PropTypes from 'prop-types'

const Button = styled.button`
  background-color: #2196f3;
  //on hover
  //on click
  color: white;
  border-radius: 2px;
  padding-left: 1vw;
  padding-right: 1vw;
  ${({ mleft }) => mleft && `margin-left: ${mleft};`}
`

Button.propTypes = {
  mleft: PropTypes.string
}

export default Button
