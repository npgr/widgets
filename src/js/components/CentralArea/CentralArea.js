import React, { useContext } from 'react'
import WidgetsContext from '../../context/WidgetsContext'
import { Block, Button } from '../StyledComponents'

export default () => {
  const screens = useContext(WidgetsContext)
  return (
    <Block
      width='50vw'
      height='90vh'
      align='center'
      bgColor='gainsboro'
      pright='0'
    >
      <Block align='right' height='10vh' pright='0'>
        <Button onClick={() => screens.addScreen('prueba')}>Save</Button>
        <Button mleft='.5vw'>Save new version</Button>
        <Button mleft='.5vw'>Submit for verification</Button>
      </Block>
      <Block bgColor='white' height='75vh'>
        {`Screen Active: ID = ${screens.activeId}`}
      </Block>
    </Block>
  )
}
