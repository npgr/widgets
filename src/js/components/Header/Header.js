import React from 'react'
import { Block } from '../StyledComponents'

export default () => <Block height='10vh'>Dashboard</Block>
