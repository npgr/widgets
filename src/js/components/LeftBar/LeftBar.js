import React from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { Block, Button } from "../StyledComponents";
import { addScreen, removeScreen } from "../../actions/screens";

const mapStateToProps = ({ screens: { screens } }) => ({
  screens
});

const mapDispatchToProps = {
  addScreen,
  removeScreen
};

const LeftBar = ({ screens, addScreen, removeScreen }) => (
  <Block width="25vw" height="90vh" pright="0">
    <Block padding="0.3vh 5px">Screens:</Block>
    <Block bgColor="#9E9E9E" padding="5px">
      <Button onClick={() => addScreen("prueba")}>+ Add</Button>
    </Block>
    <Block mtop="5vh">
      {screens.map(({ id, name }, i) => (
        <Block key={`${name}_${i}`} flex spaceBetween padding="2px">
          <Block flex padding="0">{`(${id}) ${name}`}</Block>
          <Button onClick={() => removeScreen(id)}>Delete</Button>
        </Block>
      ))}
    </Block>
  </Block>
);

LeftBar.propTypes = {
  screens: PropTypes.array.isRequired,
  addScreen: PropTypes.func.isRequired,
  removeScreen: PropTypes.func.isRequired
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(LeftBar);
