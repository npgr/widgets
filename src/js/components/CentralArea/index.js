import CentralArea from './CentralArea'

export { CentralArea }
