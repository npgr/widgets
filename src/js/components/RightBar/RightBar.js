import React from 'react'
import { useDrag } from 'react-dnd'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import {
  faChartPie,
  faChartLine,
  faChartBar,
  faChartArea
} from '@fortawesome/free-solid-svg-icons'
import { Block } from '@app/components/StyledComponents'

export default () => (
  <Block width='25vw' height='90vh' bgColor='gainsboro' ptop='4vh'>
    <Block bgColor='white' height='83vh'>
      Charts:
      <Block>
        <FontAwesomeIcon icon={faChartPie} size='3x' />
      </Block>
      <Block>
        <FontAwesomeIcon icon={faChartLine} size='3x' />
      </Block>
      <Block>
        <FontAwesomeIcon icon={faChartBar} size='3x' />
      </Block>
      <Block>
        <FontAwesomeIcon icon={faChartArea} size='3x' />
      </Block>
    </Block>
  </Block>
)
