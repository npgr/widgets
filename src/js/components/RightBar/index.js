import RightBar from './RightBar'

export { RightBar }
