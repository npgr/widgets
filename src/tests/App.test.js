import React from 'react'
import { shallow } from 'enzyme'
import App from '../js/containers/App'
import Block from '../js/components/StyledComponents/Block'
import Button from '../js/components/StyledComponents/Button'
import { Header } from '../js/components/Header'
import { LeftBar } from '../js/components/LeftBar'
import { CentralArea } from '../js/components/CentralArea'
import { RightBar } from '../js/components/RightBar'

describe('<App />', () => {
  it('renders 1 <App /> component', () => {
    const component = shallow(<App />)
    expect(component).toMatchSnapshot()
  })
})

describe('<Header />', () => {
  it('renders 1 <Header /> component', () => {
    const component = shallow(<Header />)
    expect(component).toMatchSnapshot()
  })
})

describe('<Block />', () => {
  it('renders 1 <Block /> component', () => {
    const component = shallow(<Block>Prueba</Block>)
    expect(component).toMatchSnapshot()
  })
})

describe('<Button />', () => {
  it('renders 1 <Button /> component', () => {
    const component = shallow(<Button>Prueba</Button>)
    expect(component).toMatchSnapshot()
  })
})

describe('<LeftBar />', () => {
  it('renders 1 <LeftBar /> component', () => {
    const component = shallow(<LeftBar />)
    expect(component).toMatchSnapshot()
  })
})

describe('<RightBar />', () => {
  it('renders 1 <RightBar /> component', () => {
    const component = shallow(<RightBar />)
    expect(component).toMatchSnapshot()
  })
})

describe('<CentralArea />', () => {
  it('renders 1 <CentralArea /> component', () => {
    const component = shallow(<CentralArea />)
    expect(component).toMatchSnapshot()
  })
})
