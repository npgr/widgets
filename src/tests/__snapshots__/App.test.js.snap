// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders 1 <App /> component 1`] = `
<Fragment>
  <_default />
  <styled.div
    flex={true}
    padding="0"
  >
    <Connect(LeftBar) />
    <_default />
    <_default />
  </styled.div>
</Fragment>
`;

exports[`<Block /> renders 1 <Block /> component 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
    ",
          [Function],
          "
    ",
          [Function],
          "
    ",
          [Function],
          "
    ",
          [Function],
          "
    ",
          [Function],
          "
    padding: ",
          [Function],
          ";
    ",
          [Function],
          "
    ",
          [Function],
          "
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  Prueba
</StyledComponent>
`;

exports[`<Button /> renders 1 <Button /> component 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": false,
        "rules": Array [
          "
  background-color: #2196f3;
  //on hover
  //on click
  color: white;
  border-radius: 2px;
  padding-left: 1vw;
  padding-right: 1vw;
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "styled.button",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "button",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  Prueba
</StyledComponent>
`;

exports[`<CentralArea /> renders 1 <CentralArea /> component 1`] = `
<styled.div
  align="center"
  bgColor="gainsboro"
  height="90vh"
  pright="0"
  width="50vw"
>
  <styled.div
    align="right"
    height="10vh"
    pright="0"
  >
    <styled.button>
      Save
    </styled.button>
    <styled.button
      mleft=".5vw"
    >
      Save new version
    </styled.button>
    <styled.button
      mleft=".5vw"
    >
      Submit for verification
    </styled.button>
  </styled.div>
  <styled.div
    bgColor="white"
    height="75vh"
  >
    Central Area
  </styled.div>
</styled.div>
`;

exports[`<Header /> renders 1 <Header /> component 1`] = `
<styled.div
  height="10vh"
>
  Dashboard
</styled.div>
`;

exports[`<LeftBar /> renders 1 <LeftBar /> component 1`] = `
<styled.div
  height="90vh"
  pright="0"
  width="25vw"
>
  <styled.div
    padding="0.3vh 5px"
  >
    Screens:
  </styled.div>
  <styled.div
    bgColor="#9E9E9E"
    padding="5px"
  >
    <styled.button>
      + Add
    </styled.button>
  </styled.div>
</styled.div>
`;

exports[`<RightBar /> renders 1 <RightBar /> component 1`] = `
<styled.div
  bgColor="gainsboro"
  height="90vh"
  ptop="4vh"
  width="25vw"
>
  <styled.div
    bgColor="white"
    height="83vh"
  >
    Charts
  </styled.div>
</styled.div>
`;
